src/__tests__/Dialog2.types.test.tsx

```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import {Dialog} from '../Dialog/Dialog'
+
+export function shouldAcceptCallWithNoProps() {
+  return <Dialog onClose={() => null} />
+}
+
+export function shouldNotAcceptSystemProps() {
+  // @ts-expect-error system props should not be accepted
```

Comment:
Wow, I didn't know `@ts-expect-error` was a thing. Cool!

Reply:
I love writing type tests now! Given how many regressions I've run into around types, it's nice to have some more confidence!